Add examples for each argument and improve descriptions for well swapping fm

StephanDeHoop opened the pull request:
Resolves #96 

Omitted for now the addition of copy-pasting output of `--schema` in the argument examples, since I don't think we agreed upon a final way of doing this (will address this in another PR after discussion). Added screenshots because you would need to build EVEREST docs to see the result in forward model section (with this specific branch of everest-models locally installed):
![image](https://github.com/user-attachments/assets/c670aed7-ff9d-4fdb-8660-be13d8fae168)
![image](https://github.com/user-attachments/assets/1be9a02c-430f-4130-a9d7-5c555afae74d)
![image](https://github.com/user-attachments/assets/3c36c75c-2cb8-4a4b-b0c0-559721f48fc9)
![image](https://github.com/user-attachments/assets/f967cce6-dd47-4f28-b41a-3335ef2b73f5)

edubarrosTNO commented:
LGTM, @StephanDeHoop! 🚀
Small correction spotted only: in the text describing `-cr, --constraints`, please remove the second instance of word "values" in that sentence. So it should be: "EVEREST-generated JSON file containing the values defining swapping time interval to be optimized".

Maybe someone else can review this for an unbiased second opinion?

slawekszklarz commented on src/everest_models/jobs/fm_well_swapping/cli.py:
```diff
@@ -9,6 +9,133 @@
+Argument examples
+~~~~~~~~~~~~~~~~~
+
+Example of a well swapping configuration file (:code:`-c, --config`) for a case with 4 wells, 3 time intervals and 3 possible status:
```
statuses

slawekszklarz commented on src/everest_models/jobs/fm_well_swapping/cli.py:
```diff
@@ -9,6 +9,133 @@
+Example of swapping constraints controls for a case with 3 time intervals defined in :code:`controls` section of EVEREST configuration file:
+
+.. code-block:: yaml
+
```
Í would add 'controls:' section here

Also indent is wrong, all keys should be at name level

slawekszklarz commented on src/everest_models/jobs/fm_well_swapping/cli.py:
```diff
@@ -9,6 +9,133 @@
+Example of priority controls for a case with 4 wells and 3 time intervals defined in controls section of EVEREST configuration file:
+
+.. code-block:: yaml
+
```
Í would add 'controls:' section here

slawekszklarz commented on src/everest_models/jobs/fm_well_swapping/parser.py:
```diff
@@ -36,24 +36,31 @@ def build_argument_parser(lint: bool = False, **kwargs) -> Parser:
     parser, required_group = get_parser(
-        description="Swap well operation status over multiple time intervals."
+        description=(
+            "Swaps well operation status over multiple time intervals according "
```
Here, and in lines below it is called status, in the input file it is called state, I guess state is more generic and it includes status?

slawekszklarz commented on src/everest_models/jobs/fm_well_swapping/parser.py:
```diff
@@ -36,24 +36,31 @@ def build_argument_parser(lint: bool = False, **kwargs) -> Parser:
         *_CONSTRAINTS_ARGUMENT.split("/"),
         type=_clean_constraint if not skip_type else str,
-        help="Everest generated optimized constraints",
+        help="EVEREST-generated JSON file containing the values defining each swapping time interval values to be optimized",
```
'values' mentioned twice

slawekszklarz commented on src/everest_models/jobs/shared/arguments.py:
```diff
@@ -59,7 +59,11 @@ def add_lint_argument(parser: Parser) -> None:
         "--lint",
         action="store_true",
-        help="Lints all given input (file) arguments with no data transformation.",
+        help=(
```
Help strings for linting and schema are present in majority of forward jobs, It's good to keep this strings consistent.

StephanDeHoop commented:
Addressed the review comments, can only merge once I fix the pipeline issues, feel we're close though :)! Will rebase this branch once fixed and then merge.
